libsyclinterface/include/syclinterface/dpctl_sycl_extension_interface.h

```diff
@@ -53,4 +53,22 @@ void DPCTLWorkGroupMemory_Delete(__dpctl_take DPCTLSyclWorkGroupMemoryRef Ref);
 DPCTL_API
 bool DPCTLWorkGroupMemory_Available();
 
+typedef struct RawKernelArgDataTy
+{
+    void *bytes;
```

Comment:
RawKernelArgDataTy doesn't owns `bytes` which rises a lot of questions.
Who's owning `bytes`?
Who's responsible for deleting `bytes`?
How long do we need to keep `bytes` alive?

It is probably better to make `RawKernelArgDataTy` to own `bytes` and allocate and copy data on construction of `RawKernelArgDataTy`.

Reply:
I had considered having `RawKernelArgDataTy` own the money and copy on construction. My reasoning to not do that was that the current approach allows the modification of the data in the time between construction of the raw kernel arg and the kernel launch. If you prefer to have raw kernel arg own the data, I can change that, no problem.

Reply:
I think I favor copying as well, especially because, while the overload from an object exposing buffer is pretty unambiguous (though the GC is a concern), the nbytes and pointer constructor or using the C API directly have a lot of ownership concerns.

Reply:
Okay, I switched to copying in the latest commit.